.circleci/config.yml

```diff
@@ -113,15 +113,20 @@ jobs:
             mkdir -p ~/.ssh
             ssh-keyscan github.com >> ~/.ssh/known_hosts
             VERSION=`awk -F\' '{print $2,$4}' cumulus_process/version.py`
-            git tag $VERSION
-            git push origin $VERSION
+            
+            # Only tag and release if the version doesn't already exist
+            if [ -z $(git ls-remote --tags origin | grep $VERSION) ]; then
+              git tag $VERSION
+              echo "Pushing tag ${VERSION}"
+              git push origin $VERSION
+            fi
       - run:
           name: Deploy to PyPi
           command: |
             . ~/venv36/bin/activate
             pip install twine
             python setup.py sdist
-            twine upload --username "${PYPI_USER}" --password "${PYPI_PASS}" dist/*
+            twine upload --skip-existing --username "${PYPI_USER}" --password "${PYPI_PASS}" dist/*
```

Comment:
One thought - have we considered tagging post master build (if updated) , then using something like 

https://circleci.com/docs/2.0/workflows/#git-tag-job-execution  to trigger deployments?

Reply:
I have used it before. last time I used it it wasn't stable enough, meaning that the workflow step didn't trigger on a new tags most of the times. Not sure why.

Reply:
also I personally prefer idempotent steps that can be run as many times without breaking things.

Reply:
Interesting, I haven't seen that instability elsewhere.     In either case this also LGTM.